### ISSUE TITLE
Remember resized maximized panel's height

### ISSUE THREAD

**Issue by octref** [microsoft/vscode#66303]
Currently, I get 30% height terminal when I do `` Ctrl + ` ``, and 90% height terminal when I do "Toggle Maximize Terminal".

If i try to resize the panel, I only get to change the height of the "default height" for terminal.

Let's say:

- I run "Toggle Maximized Terminal"
- I resize it to 60% height
- "Toggle Maximized Terminal" should toggle between close state and 60%
- "Toggle Terminal" should toggle between close state and 30%

However, current behavior is "Toggle Maxmized Terminal" will always go to 90% height, which I don't want.

**Comment by sbatten** [comment]
@octref `Toggle Maximized Panel` toggles between maximized and last known size. This is the same as windowing maximization. If you want to toggle between 60% and closed, you should be using `Toggle Terminal`

**Comment by octref** [comment]
I want 0, 30% and 60%. Currently there's no way to do that.

"Maxmized" as 90% is not really usable for me

**Comment by tomasaschan** [comment]
I think I'm seeing the behavior described here as well, and to me it's also confusing. What's happening is the terminal seems to forget its previous size if I try to toggle between three states: closed, maximized and a third size (in my case usually around 30-50%). Here's a gif that shows the behavior:

![terminal-resize](https://user-images.githubusercontent.com/1550920/64960774-3e178900-d894-11e9-84c0-4cb756b10e2a.gif)

What I'm doing in this screen recording is the following:

1. Open a VS Code window with a terminal pane sized at around 30%.
1. Toggle the terminal a couple of times. It remembers the size it was, and goes back and forth between that size.
1. Toggle between normal and maximized a couple of times. It remembers again.
1. From maximized, toggle to closed and back. When it comes back, it's not maximized, but at 90%, and my previous size is now forgotten and there's no way to get back to it without using my mouse.

In other words, the problem seems to be that closing a maximized terminal sets the "previous size" to 90%, instead of whatever it was before the panel was maximied.

If this is a different problem from the one you tried to describe, @octref, let me know and I'll file a separate issue for it.

**Comment by sbatten** [comment]
@tomasaschan the issue you are reporting is covered by #80366

**Comment by tomasaschan** [comment]
@sbatten Thanks; I'll keep a lookout for a fix there then!
